game-core/src/main/java/games/strategy/engine/framework/startup/ui/posted/game/pbf/ForumPosterEditor.java

```diff
@@ -63,6 +65,10 @@
     super(new GridBagLayout());
     this.viewModel = viewModel;
     viewModel.setView(this);
+    // If password is already stored we already remember
+    if (viewModel.isForumPasswordValid()) {
```

Comment:
nit, view should not be doing logic if it has a view model. The view model should be the one that owns the full logic of when 'rememberPassword' is selected.

Would recommend to fix by moving the logic to view model, EG:
```
rememberPassword.setSelected(viewModel.isRememberPasswordSelected());
```

Reply:
To be honest this is more like a hack.
If the user already has a token stored, it makes no sense to have "Remember Password" unchecked, which would cause the credentials to be removed on close by default.
On the other hand if no token is stored it "defaults" to the ClientSetting that stores the last state. That's why the code is the way it is.

I think "Remember Password" shouldn't be a client setting at all. The checkbox should be selected by default if and only if the token is already stored and leave it up to the user to alter this behaviour.

Reply:
The 'remember password' is a setting as we are currently supporting passwords, and if you select the box then you want it checked, and if you unselect it then it'll be annoying to have to unselect it every time. Those users are probably very worried about their password and if they forget to unselect the value then they'll be unhappy with the software for allowing the error.

It sounds like if 'remember password' does not apply with tokens then it should be hidden.

Either way, any `if` statements should be removed from the view and moved to the model. The view should not know how to compute the UI state, it should ask the model what its appropriate state should be and render that.